File: beware-of-the-sheep/src/Entities/Plants/Guarana.cpp
#include "pch.h"
#include "Guarana.h"
#include "World.h"
#include "Entities/Animals/Animal.h"

namespace bots {

	Guarana::Guarana(World & world, const Point & position) :
		Plant(world, position)
	{}

	void Guarana::collision(Entity & other)
	{
		if (dynamic_cast<Animal*>(&other)) {
			other.addStrength(3);
			std::string message = other.getClassName() + " has now " + std::to_string(other.getStrength()) + " Strength";
			m_world.addMessage(std::move(message));
		}
	}

} // namespace bots

File: src/com/kostu96/bots/entities/plants/Guarana.java
package com.kostu96.bots.entities.plants;

import com.kostu96.bots.World;
import com.kostu96.bots.entities.Entity;
import com.kostu96.bots.entities.animals.Animal;
import com.kostu96.bots.utils.ImageManager.ImageID;
import com.kostu96.bots.utils.Point;

public class Guarana extends Plant {
	private static final int STRENGTH = 0;

	public Guarana(World world, Point position) {
		super(world, position, STRENGTH);
	}

	@Override
	public void collision(Entity other) {
		if (other instanceof Animal) {
			other.addStrength(3);
			//std::string message = other.getClassName() + " has now " + std::to_string(other.getStrength()) + " Strength";
			//m_world.addMessage(std::move(message));
		}
	}

	@Override
	public ImageID getImageID() {
		return ImageID.GUARANA;
	}

}
